src/main/java/com/scmp/framework/testng/listeners/SuiteListener.java

```diff
@@ -180,4 +190,91 @@ private void createNewTestRun(ISuite suite, String projectId, String testRunName
+		LocalDate today = LocalDate.now(runTimeContext.getZoneId());
+		String timestamp = String.valueOf(today.minusDays(frameworkConfigs.getFailedCaseTestRunWithinDays()).atStartOfDay(runTimeContext.getZoneId()).toEpochSecond());
+		HashSet<Integer> failedTestCasesIdSet = new HashSet<>();
+		boolean isFirstRunLoop = true;
+		Map<Integer, String> finalResultMap = new HashMap<>();
+
+		try {
+
+			Pattern matchTestRunPattern = Pattern.compile(frameworkConfigs.getFailedCaseTestRunNotificationPattern());
+
+			// Get the latest x test runs and match with pattern set in configuration
+			List<TestRun> runs = testRailManager.getTestRuns(projectId, timestamp).getTestRunList().stream()
+					.filter(testRun -> matchTestRunPattern.matcher(testRun.getName()).find())
+					.limit(frameworkConfigs.getFailedCaseNotificationCount())
+					.toList();
+
+			// Search for consecutive failed test cases
+			for (TestRun run : runs) {
+
+				// Skip listing if the test run is in progress
+				if (!testRailManager.getAllTestRunTests(run.getId(), String.valueOf(TestRailStatus.IN_PROGRESS)).isEmpty()) {
+					return;
+				}
+
+				// Get all failed test cases
+				Map<Integer, String> runResult = testRailManager.getAllTestRunTests(run.getId(), String.valueOf(TestRailStatus.Failed))
+						.stream()
+						.collect(Collectors.toMap(TestRunTest::getCaseId, TestRunTest::getTitle));
+
+				if (isFirstRunLoop) {
+					failedTestCasesIdSet.addAll(runResult.keySet());
+					finalResultMap.putAll(runResult);
+					isFirstRunLoop = false;
+				} else {
+					failedTestCasesIdSet.retainAll(runResult.keySet());
+				}
+			}
+
+			// Remove those test that are failed first time only but not consecutive times
+			finalResultMap.keySet().removeIf(testId -> !failedTestCasesIdSet.contains(testId));
+
+			// Remove test case in exclude list
+			HashSet<String> excludeTestCase = new HashSet<>(Arrays.asList(frameworkConfigs.getFailedCaseExcludeList().split(",")));
+			finalResultMap.keySet().removeIf(testId -> excludeTestCase.contains(String.valueOf(testId)));
+
+			// Display list
+			if (!finalResultMap.isEmpty()) {
+				frameworkLogger.info("Consecutive failed test cases: ");
+				finalResultMap.forEach((k, v) -> frameworkLogger.info("Test Case ID: {}, Title: {}", k, v));
+			} else {
+				frameworkLogger.info("No consecutive failed test cases found.");
+			}
+
+			// Send message to Slack
+			AtomicReference<String> message = new AtomicReference<>("");
+			String notifyUserMessage;
+
+			if(!frameworkConfigs.getFailedCaseNotificationSlackUserId().isEmpty()){
+				String notifyUserMessageFormat = "Hi <@%s>, please find the following test cases that failed for %s consecutive runs in test run: <%sindex.php?/runs/view/%s|%s>\n";
+				notifyUserMessage = String.format(notifyUserMessageFormat, frameworkConfigs.getFailedCaseNotificationSlackUserId(), frameworkConfigs.getFailedCaseNotificationCount(), frameworkConfigs.getTestRailServer(), runs.get(0).getId(), runs.get(0).getName());
```

Comment:
let's try to reduce redundancy by using variables
```
String userId = frameworkConfigs.getFailedCaseNotificationSlackUserId();
int notificationCount = frameworkConfigs.getFailedCaseNotificationCount();
String testRailServer = frameworkConfigs.getTestRailServer();
String runId = runs.get(0).getId();
String runName = runs.get(0).getName();
```

Reply:
Separated variables and added to the move of the function